### ISSUE TITLE
Prevent chainsaw from spawning on Beginner/Easy action levels

### ISSUE THREAD

**Issue by Bigmanjapan** [elishacloud/Silent-Hill-2-Enhancements#138]
@Polymega

In a clean game build (with no Game Result savefiles which make every new walkthrough to have a NG+ property) this code is responsible for spawning the wooden logs with chainsaw: 

```
0048AB68 cmp byte ptr [01DBBFF4],02
0048AB6F jae 0048AB80
0048AB71 mov eax,[01F7A9E0]
0048AB76 or eax,1000
0048AB7B mov [01F7A9E0],eax
0048AB80 test esi,esi
```

sh2pc.exe+19BBFF4 (01DBBFF4) byte — Action level (0 = beginner, 1 = easy, 2 = normal, 3 = hard)
sh2pc.exe+1B7A9E0 (01F7A9E0) 4 byte — some internal value that controls chainsaw spawning (if the lower 2 bytes are set to 0x0000 manually, then the chainsaw won't spawn)

The code above checks if the Action level equals or greater than 2 (cmp + jae instructions). If that's the case, then it skips next three lines and proceeds to 0048AB80.

Changing 02 to 00 (switching normal to beginner) in the first line prevents game engine from spawning the chainsaw on all difficulties if the game was not completed before (as it should be). 

I wrote "In a clean game build" at the beginning because it seems that in a NG+ condition a different code is responsible for spawning the chainsaw and it works correctly despite the above modification. Meaning it's only needed to change one byte from 02 to 00 to solve the issue entirely. 

A said byte is located at 0х8AB6E in sh2pc.exe.

My main concern is that the skipped few lines could be responsible for something else Action level related like monsters' HPs although highly unlikely. If it turns out that there are actually issues, then it is possible to null the chainsaw spawning value prior to entering the chainsaw location (while being at the cemetery, for example) and then set the value back upon leaving the chainsaw location. Right now this method has an issue of the chainsaw sound still being present but I can find how to kill it if needs be.

**Comment by Polymega** [comment]
While not as bad as receiving the UFO ending on your first play through like with _Homecoming_, this is still a great thing to fix and make right. I'm glad you caught this and found a solution.

**Comment by Polymega** [comment]
I changed the instruction to `0048AB68 cmp byte ptr [01DBBFF4],00` and didn't hear any chainsaw noise on a clean new game. That's good news. Can you test it on your side one more time?

**Comment by Polymega** [comment]
Thank you again Bigmanjapan for this fix. Elisha, when you're ready, let's try the following please: 

On game launch, at `0048AB68`, change instruction `cmp byte ptr [01DBBFF4],02` to `cmp byte ptr [01DBBFF4],00`.

@Bigmanjapan when Elisha implements this fix may I ask you to help me play test this one? An extra set of eyes will help with testing this new feature. 

@elishacloud after this one all that's left is implementing the [post processing fixes](https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/66#issuecomment-519684197) before we can jump over to shadow work--which I'm very pumped for!

**Comment by elishacloud** [comment]
Ok, I finished implementing this.  Test module: [d3d8.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/3603839/d3d8.zip)

**Comment by Polymega** [comment]
I believe we're good to go with this one. I did full to near-full playthroughs on all action difficulties using the 1.0 binary. Enemy HP, speed, behavior/aggressiveness, and spawn rates all seemed correct. I then did quick runs up to where the chainsaw should/does spawn on the 1.1 and DC binaries as well, and everything seemed fine.

**Comment by elishacloud** [comment]
This has been implemented in ce22617bfe45be3329bf99d4d1eeca95d37f4e6a.  Closing.
